### DISCUSSION TITLE
求解：走漏网之鱼的dns泄露网站为什么依旧向namesever国内组dns发起请求？

### DISCUSSION THREAD

**Discussion by ZekTy** [vernesong/OpenClash#4178] [Q&A]
使用meta内核，混合模式，DNS设置勾选自定义上游dns服务器，勾选追加上游dns，使用运行商下发的dns解析国内域名，namesever组不勾选；fallback组不勾选，希国外域名发送到节点做dns解析。开启允许ipv6解析，不开启代理；勾选绕过中国大陆域名。
在yacd面板中可以看到browserleaks.com的域名全都走了漏网之鱼代理，如下：
e1a8k8v3nsdwjiu9.dns6.browserleaks.org:443 |   | 3.61 KB | 597 B | 0 B/s | 0 B/s | 🐟 漏网之鱼 / 🇹🇼 台湾节点 / 🇨🇳 台湾标准 IEPL 中继 3 | Match | less than a minute | 192.168.1.188:27685 | 104.236.69.55 | Redir(tcp)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
l9tw9s7qjzjzmquu.dns6.browserleaks.org:443 |   | 3.71 KB | 598 B | 0 B/s | 0 B/s | 🐟 漏网之鱼 / 🇹🇼 台湾节点 / 🇨🇳 台湾标准 IEPL 中继 3 | Match | less than a minute | 192.168.1.188:27644 | 104.236.69.55 | Redir(tcp)
035ib0z444hj87t9.dns4.browserleaks.org:443 |   | 3.63 KB | 597 B | 0 B/s | 0 B/s | 🐟 漏网之鱼 / 🇹🇼 台湾节点 / 🇨🇳 台湾标准 IEPL 中继 3 | Match | less than a minute | 192.168.1.188:27686 | 104.236.69.55 | Redir(tcp)
......
结果网站仍然检测出了国内的dns域名，在手动取消勾选“追加上游dns”，勾选namesever组dns为8.8.8.8时网站检测不到国内的ip
故判断是走代理的域名仍向namesever组发起了请求，包括像ipleak这个网站也是走的漏网之鱼也依旧检测出了国内ip，有理由怀疑不包含在规则集之内的小众国内外网站，都可能产生dns泄露，使用过两个订阅转换模板都是一样的情况，一个是ACL4SSR的全分组full，一个是[Custom_OpenClash_Rules](https://github.com/Aethersailor/Custom_OpenClash_Rules)。以下分别是两个模板漏网之鱼部分的规则：
  - PROCESS-NAME,DownloadService.exe,🎯 全球直连
  - PROCESS-NAME,Weiyun.exe,🎯 全球直连
  - PROCESS-NAME,baidunetdisk.exe,🎯 全球直连
  - GEOIP,CN,🎯 全球直连
  - MATCH,🐟 漏网之鱼
 ——————
  - RULE-SET,ChinaIp,🎯 全球直连
  - RULE-SET,Download,🎯 全球直连
  - GEOSITE,category-public-tracker,🎯 全球直连
  - GEOIP,LAN,🎯 全球直连,no-resolve
  - GEOIP,CN,🎯 全球直连,no-resolve
  - MATCH,🐟 漏网之鱼

想请教openclash dns分流规则的原理和出现这种情况的原因
用的是最新的istoreos，iptables。[v0.46.050-beta]最新版本的openclash

**Comment by fejeIaqbhbv** [comment]
没有设policy吧？
那就是设计预期流程
https://wiki.metacubex.one/config/dns/diagram
